### PR TITLE
BUG: rolling.count with axis=1 

### LINKED ISSUE #13503: Rolling(axis='columns').count() ignores axis= keyword
addtl example on #13753 
#### Code Sample, a copy-pastable example if possible

``` python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame({'x': range(5), 'y': range(5)})

In [3]: df
Out[3]: 
   x  y
0  0  0
1  1  1
2  2  2
3  3  3
4  4  4

In [4]: df.rolling(2, axis='columns').sum()  # meets expectations
Out[4]: 
    x    y
0 NaN  0.0
1 NaN  2.0
2 NaN  4.0
3 NaN  6.0
4 NaN  8.0

In [5]: df.rolling(2, axis='columns').count()  # appears to be the same as axis='rows'
Out[5]: 
     x    y
0  1.0  1.0
1  2.0  2.0
2  2.0  2.0
3  2.0  2.0
4  2.0  2.0

In [6]: df.rolling(2, axis='rows').count()  # yeah, exactly the same
Out[6]: 
     x    y
0  1.0  1.0
1  2.0  2.0
2  2.0  2.0
3  2.0  2.0
4  2.0  2.0
```
#### output of `pd.show_versions()`

``` python
In [7]: pd.show_versions()

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-24-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.1
nose: 1.3.7
pip: 8.1.1
setuptools: 20.3
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.17.0
statsmodels: 0.6.1
xarray: None
IPython: 4.1.2
sphinx: 1.3.5
patsy: 0.4.0
dateutil: 2.5.1
pytz: 2016.2
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: 2.3.2
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.8.4
lxml: 3.6.0
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.39.0
pandas_datareader: None
```

Arose from tests in https://github.com/dask/dask/pull/1280

### PR DESCRIPTION
**Pull request by yhaque1213**
- [ ] closes #13503 
- [X] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry
Added axis parameter to the constructor call in the count function, fixing the original issue in #13503. This fix works for the example given in the issue. This does not fix the problem addressed by jorisvandenbossche, as that seems to be a type issue.

### CODE REVIEW

**Comment by jreback** [comment]
tests pls

File: pandas/tests/test_window.py [hunk 1]
```diff
@@ -667,6 +667,18 @@ def test_rolling_axis(self, axis_frame):
...
+        # see gh-26055
+        df = pd.DataFrame({'x':range(5), 'y':range(5)})
+
+        df_correct_row_count = pd.DataFrame({'x':[1.0, 2.0, 2.0, 2.0, 2.0], 'y': [1.0, 2.0, 2.0, 2.0, 2.0]})
+
+        assert_equal(df.rolling(2).count(), df_correct_row_count)
+        assert_equal(df.rolling(2, axis='rows').count(), df_correct_row_count)
+
+        df_correct_column_count = pd.DataFrame({'x':[1.0, 1.0, 1.0, 1.0, 1.0], 'y': [2.0, 2.0, 2.0, 2.0, 2.0]})
+        assert_equal(df.rolling(2, axis='columns').count(), df_correct_column_count)
```
Comment (gfyoung) [inline pandas/tests/test_window.py]: Use `tm.assert_frame_equal` instead of `assert_equal`.

**Comment by yhaque1213** [comment]
@gfyoung 
I updated the tests to replace my calls to `assert_equal` with `tm.assert_frame_equal`, and I also modified it to pass pep8 checks. My initial changes to fix the issue, without any tests, was passing all of the builds.

Once I added my tests, however, I started failing pandas-dev.pandas and pandas-dev.pandas (Linux py35_compat), and I'm having some trouble figuring out what's causing this to happen.

More specifically, it seems to be an error with the `test_drop_duplicates_categorical_non_bool` in `pandas/tests/series/test_analytics.py`, but I have no idea why this specific test would be failing because of adding my test, `test_count_axis`. 

Do you have any advice?

**Comment by gfyoung** [comment]
@yhaque1213 : Have you rebased onto the most recent `master`?  Try that and see if it goes away.  Otherwise, we'll look into seeing if we can remove (or weaken) the `xfail` on those tests.

**Comment by yhaque1213** [comment]
@gfyoung : that seemed to do the trick - my latest update is passing all of the checks. Thank you, and please let me know if there's anything else you'd like me to add!

**Comment by gfyoung** [comment]
@yhaque1213 : Don't forget the `whatsnew` entry

File: pandas/tests/test_window.py [hunk 2]
```diff
@@ -667,6 +667,18 @@ def test_rolling_axis(self, axis_frame):
         result = df.rolling(3, axis=axis_frame).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_count_axis(self):
+        # see gh-26055
+        df = DataFrame({'x': range(3), 'y': range(3)})
+        df_row = DataFrame({'x': [1.0, 2.0, 2.0], 'y': [1.0, 2.0, 2.0]})
+
+        # not specifiying axis and making axis=rows should be the same result
+        tm.assert_frame_equal(df.rolling(2).count(), df_row)
```
Comment (jreback) [inline pandas/tests/test_window.py]: use 

result=
expected=
tm.assert_frame_equal

so df_row looks like your expected

File: pandas/tests/test_window.py [hunk 3]
```diff
@@ -667,6 +667,18 @@ def test_rolling_axis(self, axis_frame):
...
+        tm.assert_frame_equal(df.rolling(2, axis='rows').count(), df_row)
+
+        df_col = DataFrame({'x': [1.0, 1.0, 1.0], 'y': [2.0, 2.0, 2.0]})
```
Comment (jreback) [inline pandas/tests/test_window.py]: same here

File: pandas/tests/test_window.py [hunk 4]
```diff
@@ -667,6 +667,18 @@ def test_rolling_axis(self, axis_frame):
         result = df.rolling(3, axis=axis_frame).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_count_axis(self):
```
Comment (jreback) [inline pandas/tests/test_window.py]: rename this to test_rolling_axis_count, rename the test right above to test_rolling_axis_sum

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatsnew note as well; bug fixes groupby/rolling section

File: doc/source/whatsnew/v0.25.0.rst [hunk 5]
```diff
@@ -384,6 +384,7 @@ Groupby/Resample/Rolling
 - Ensured that result group order is correct when grouping on an ordered ``Categorical`` and specifying ``observed=True`` (:issue:`25871`, :issue:`25167`)
 - Bug in :meth:`pandas.core.window.Rolling.min` and :meth:`pandas.core.window.Rolling.max` that caused a memory leak (:issue:`25893`)
 - Bug in :func:`idxmax` and :func:`idxmin` on :meth:`DataFrame.groupby` with datetime column would return incorrect dtype (:issue:`25444`, :issue:`15306`)
+- Bug in :meth:`pandas.core.window._Rolling_and_Expanding.count` that did not include axis parameter in constructor call (:issue:`13503`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this is not a public method, use ``:meth:`pandas.core.window.Rolling.count`` and another for Expanding

Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: this is also confusing over what you are saying, can you reword, something like '....was previously ignoring the ``axis`` keyword'

File: pandas/tests/test_window.py [hunk 6]
```diff
@@ -667,6 +667,23 @@ def test_rolling_axis(self, axis_frame):
         result = df.rolling(3, axis=axis_frame).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_rolling_axis_count(self):
```
Comment (jreback) [inline pandas/tests/test_window.py]: use the axis_frame fixture which resoles to rows/columns/0/1

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by yhaque1213** [comment]
I made the changes to the test and to the docs, but it seems like these Azure pipelines checks are passing arbitrarily. For instance, after fixing a typo in a comment (commit #12065b6) we passed 10/10 tests, while the previous commit only passed 8 (commit #41ae7fe). This happened even with rebasing before each push, therefore, we aren't sure what to change for this last commit to get it to pass all 10.

**Comment by jreback** [comment]
> I made the changes to the test and to the docs, but it seems like these Azure pipelines checks are passing arbitrarily. For instance, after fixing a typo in a comment (commit #12065b6) we passed 10/10 tests, while the previous commit only passed 8 (commit #41ae7fe). This happened even with rebasing before each push, therefore, we aren't sure what to change for this last commit to get it to pass all 10.

CI is not 'arbitrary', rather there was an upstream change that broke things.

**Comment by jreback** [comment]
pls merge master and ping on green.

File: doc/source/whatsnew/v0.25.0.rst [hunk 7]
```diff
@@ -384,6 +384,8 @@ Groupby/Resample/Rolling
 - Ensured that result group order is correct when grouping on an ordered ``Categorical`` and specifying ``observed=True`` (:issue:`25871`, :issue:`25167`)
 - Bug in :meth:`pandas.core.window.Rolling.min` and :meth:`pandas.core.window.Rolling.max` that caused a memory leak (:issue:`25893`)
 - Bug in :func:`idxmax` and :func:`idxmin` on :meth:`DataFrame.groupby` with datetime column would return incorrect dtype (:issue:`25444`, :issue:`15306`)
+- Bug in :meth:`pandas.core.window.Rolling.count` was previously ignoring the axis keyword (:issue:`13503`)
+- Bug in :meth:`pandas.core.window.Expanding.count` was previously ignoring the axis keyword (:issue:`13503`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: make a single entry here (just combine the lines)

File: pandas/tests/test_window.py [hunk 8]
```diff
@@ -667,6 +667,21 @@ def test_rolling_axis(self, axis_frame):
         result = df.rolling(3, axis=axis_frame).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_rolling_axis_count(self, axis_frame):
+        # see gh-26055
+        df = DataFrame({'x': range(3), 'y': range(3)})
+
+        axis = df._get_axis_number(axis_frame)
+
+        if axis == 0:
```
Comment (jreback) [inline pandas/tests/test_window.py]: this is not correct (should your test have broken?)

``if axis in [0, 'index']`` is what you need

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by yhaque1213** [comment]
@jreback : Just made the changes requested, and I'm passing all the checks now!

File: doc/source/whatsnew/v0.25.0.rst [hunk 9]
```diff
@@ -390,6 +390,8 @@ Groupby/Resample/Rolling
 - Ensured that ordering of outputs in ``groupby`` aggregation functions is consistent across all versions of Python (:issue:`25692`)
 - Ensured that result group order is correct when grouping on an ordered ``Categorical`` and specifying ``observed=True`` (:issue:`25871`, :issue:`25167`)
 - Bug in :meth:`pandas.core.window.Rolling.min` and :meth:`pandas.core.window.Rolling.max` that caused a memory leak (:issue:`25893`)
+- Bug in :func:`idxmax` and :func:`idxmin` on :meth:`DataFrame.groupby` with datetime column would return incorrect dtype (:issue:`25444`, :issue:`15306`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: looks like this was picked up somehow, pls remove

**Review by jreback — CHANGES_REQUESTED** [review]
looks good; edit the whatsnew; ping on green.

**Comment by yhaque1213** [comment]
I've made all of the changes that were requested, but I'm still having trouble getting all of the checks to pass. I've tried rebasing and merging multiple times, but nothing has been working

**Comment by jreback** [comment]
try

git merge upstream/master

**Comment by WillAyd** [comment]
There's a failure in the doc checks that should be unrelated. Not sure if its a versioning thing or a commit to master. Taking a look now

**Comment by jreback** [comment]
this was cleared yesterday

**Comment by yhaque1213** [comment]
Latest push was with merging with upstream/master, but it looks like the checks are still failing in the same ways as before. Sorry I'm not sure if I'm doing something wrong, but I've tried everything I could think of

**Review by jreback — APPROVED** [review]

File: doc/source/whatsnew/v0.25.0.rst [hunk 10]
```diff
@@ -387,6 +387,7 @@ Groupby/Resample/Rolling
 - Ensured that ordering of outputs in ``groupby`` aggregation functions is consistent across all versions of Python (:issue:`25692`)
 - Ensured that result group order is correct when grouping on an ordered ``Categorical`` and specifying ``observed=True`` (:issue:`25871`, :issue:`25167`)
 - Bug in :meth:`pandas.core.window.Rolling.min` and :meth:`pandas.core.window.Rolling.max` that caused a memory leak (:issue:`25893`)
+- Bug in :meth:`pandas.core.window.Rolling.count` and `pandas.core. window.Expanding.count` was previously ignoring the axis keyword (:issue:`13503`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: can you remove the space here pandas.core.window....

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. tiny typo. pls rebase on master which is green now.

**Comment by jreback** [comment]
small typo plus can i merge master ; ping on green

**Comment by yhaque1213** [comment]
@jrebeck: I removed the typo - I'm passing all the checks now!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @yhaque1213

**Pull request merged**